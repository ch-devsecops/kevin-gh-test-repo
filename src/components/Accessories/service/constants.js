import PropTypes from 'prop-types';

export const variant1 = 'variant1';
export const variant2 = 'variant2';
export const variant3 = 'variant3';

export const ACCESSORY_TYPE_LABEL_ALL = 'all';
export const ALL_TAB = 'all';
export const INTERIOR_TAB = 'interior';
export const EXTERIOR_TAB = 'exterior';
export const PACKAGES_TAB = 'packages';
export const TRIM_DROPDOWN_KEY = 'trim';
export const TRANSMISSION_DROPDOWN_KEY = 'transmission';
export const ACCESSORY_TYPE_DROPDOWN_KEY = 'accessoryType';
export const ACCESSORIES_WRAPPER_INTERACTION_TYPE = 'accessories - accessory filter';
export const ACCESSORIES_CARD_INTERACTION_TYPE = 'accessory details';

export const gtmType = PropTypes.shape({
  'data-gtm-model': PropTypes.string,
  'data-gtm-trim': PropTypes.string,
  'data-gtm-body-style': PropTypes.string,
});
