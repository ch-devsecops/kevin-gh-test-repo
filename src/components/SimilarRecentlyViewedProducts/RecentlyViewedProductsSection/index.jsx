import React, { useContext, useEffect, useState } from 'react';
import PropTypes from 'prop-types';
import { useTranslation } from 'react-i18next';
import { H4 } from '@honda-canada/design-system-react';
import GridSection from '../../ProductCardsGrid/GridSection';
import LocalStorageContext from '../../LocalStorageContext/LocalStorageContext';

const Header = ({ name }) => (
  <H4 mt={['m', 'default']} aria-label={name}>
    {name}
  </H4>
);
Header.propTypes = { name: PropTypes.string };

const RecentlyViewedProductsSection = ({ models = [], gtmTags, getModels, fetchFinancial, ...rest }) => {
  const { t } = useTranslation();
  const [modelsList, setModelsList] = useState([]);
  const { recentlyViewedProducts } = useContext(LocalStorageContext);

  useEffect(() => {
    setModelsList(getModels(recentlyViewedProducts, models));
  }, [recentlyViewedProducts]);

  if (!models?.length) return null;

  return (
    <GridSection
      models={modelsList}
      gridTemplateColumnsCount={modelsList.length}
      header={<Header name={t('Pages.Models.Exploration.recentlyViewedProductsLabel')} />}
      textAlign="center"
      display="flex"
      flexDirection="column"
      width={[1, 1, 1 / 2]}
      gtmTags={gtmTags}
      fetchFinancial={fetchFinancial}
      {...rest}
    />
  );
};

RecentlyViewedProductsSection.propTypes = {
  models: PropTypes.arrayOf(PropTypes.shape({})),
  fetchFinancial: PropTypes.bool,
  getModels: PropTypes.func,
  gtmTags: PropTypes.shape({
    componentName: PropTypes.string,
    title: PropTypes.string,
    compareTitle: PropTypes.string,
    interactionType: PropTypes.string,
    compareInteractionType: PropTypes.string,
  }),
};

export default RecentlyViewedProductsSection;
