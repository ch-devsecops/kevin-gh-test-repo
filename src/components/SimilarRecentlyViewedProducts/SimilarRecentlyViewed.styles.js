import css from '@styled-system/css';
import { ThemeStyles } from '@honda-canada/design-system-react';

const themeStyles = new ThemeStyles();

themeStyles.addBaseStyles('Container')(({ margins, flexDesktopAlignment }) =>
  css({
    display: 'flex',
    flexDirection: ['column', 'column', 'row'],
    justifyContent: ['space-around', 'space-around', flexDesktopAlignment],
    flexWrap: 'wrap',
    mt: [
      margins?.topMarginMob ? margins?.topMarginMob : 'xl',
      margins?.topMarginMob ? margins?.topMarginMob : 'xl',
      margins?.topMargin ? margins?.topMargin : 'big',
    ],
    mb: [
      margins?.bottomMarginMob ? margins?.bottomMarginMob : 'xl',
      margins?.bottomMarginMob ? margins?.bottomMarginMob : 'xl',
      margins?.bottomMargin ? margins?.bottomMargin : 'big',
    ],
    mx: [
      margins?.horzMarginMob ? margins?.horzMarginMob : 'zero',
      margins?.horzMarginMob ? margins?.horzMarginMob : 'zero',
      margins?.horizontalMargin ? margins?.horizontalMargin : 'auto',
    ],
  }),
);

export default themeStyles;
