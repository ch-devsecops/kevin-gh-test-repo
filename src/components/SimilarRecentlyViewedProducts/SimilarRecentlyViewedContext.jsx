import React, { useState, useMemo } from 'react';

const SimilarRecentlyViewedContext = React.createContext({});
SimilarRecentlyViewedContext.displayName = 'SimilarRecentlyViewedContext';

export default SimilarRecentlyViewedContext;

export const SimilarRecentlyViewedProvider = ({ children }) => {
  const [minBodyTextHeight, setMinBodyTextHeight] = useState('0');
  const memoisedValues = useMemo(
    () => ({
      minBodyTextHeight,
      setMinBodyTextHeight,
    }),
    [minBodyTextHeight, setMinBodyTextHeight],
  );

  return (
    <SimilarRecentlyViewedContext.Provider value={memoisedValues}>{children}</SimilarRecentlyViewedContext.Provider>
  );
};
