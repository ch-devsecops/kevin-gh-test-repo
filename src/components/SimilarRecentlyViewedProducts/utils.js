/**
 * flatten the nest properties of engine models
 * @param {array<string>}detIds
 * @param {array<string>}data
 * @returns {*[]}
 */
export function getEngineModels(detIds, data) {
  const ret = [];
  detIds?.forEach(detId => {
    data?.forEach(category => {
      category?.crankshafts?.forEach(crankshaft => {
        const match = crankshaft?.models?.find(
          model => model?.detIdentifier?.value === detId && !ret?.find(item => item?.detIdentifier?.value === detId),
        );

        if (match) {
          match.seriesName = category.name;
          match.crankshaftName = crankshaft.name;
          ret.push(match);
        }
      });
    });
  });
  return ret;
}

/**
 * flatten the nest properties of marine models
 * @param detIds
 * @param data
 * @returns {*[]}
 */
export function getMarineModels(detIds, data) {
  const ret = [];
  detIds?.forEach(detId => {
    data?.forEach(category => {
      const match = category?.models?.find(
        model => model?.detIdentifier?.value === detId && !ret?.find(item => item?.detIdentifier?.value === detId),
      );

      if (match) {
        match.seriesName = category.name;
        match.crankshaftName = category.models.name;
        ret.push(match);
      }
    });
  });
  return ret;
}
