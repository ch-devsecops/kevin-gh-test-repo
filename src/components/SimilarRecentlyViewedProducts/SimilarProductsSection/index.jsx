import React from 'react';
import PropTypes from 'prop-types';
import { H4 } from '@honda-canada/design-system-react';
import { useTranslation } from 'react-i18next';
import GridSection from '../../ProductCardsGrid/GridSection';
import { SIMILAR_PRODUCTS_LIMIT } from '../../../utils/constants';

const Header = ({ name }) => (
  <H4 mt={['m', 'default']} aria-label={name}>
    {name}
  </H4>
);
Header.propTypes = { name: PropTypes.string };

const SimilarProductsSection = ({ models = [], gtmTags, fetchFinancial, ...rest }) => {
  const { t } = useTranslation();
  const modelItemLength = models?.length;

  if (modelItemLength === 0) return null;

  return (
    <GridSection
      models={models?.slice(0, SIMILAR_PRODUCTS_LIMIT)}
      gridTemplateColumnsCount={modelItemLength > SIMILAR_PRODUCTS_LIMIT ? SIMILAR_PRODUCTS_LIMIT : modelItemLength}
      header={<Header name={t('Pages.Models.Exploration.similarProductsLabel')} />}
      textAlign="center"
      display="flex"
      flexDirection="column"
      width={[1, 1, 1 / 2]}
      gtmTags={gtmTags}
      fetchFinancial={fetchFinancial}
      {...rest}
    />
  );
};

SimilarProductsSection.propTypes = {
  models: PropTypes.arrayOf(PropTypes.shape({})),
  fetchFinancial: PropTypes.bool,
  gtmTags: PropTypes.shape({
    componentName: PropTypes.string,
    title: PropTypes.string,
    compareTitle: PropTypes.string,
    interactionType: PropTypes.string,
    compareInteractionType: PropTypes.string,
  }),
};

export default SimilarProductsSection;
