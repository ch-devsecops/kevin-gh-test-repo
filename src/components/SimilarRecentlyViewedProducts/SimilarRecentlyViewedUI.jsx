import React from 'react';
import PropTypes from 'prop-types';
import { Box, Wrapper } from '@honda-canada/design-system-react';
import SimilarProductsSection from './SimilarProductsSection';
import RecentlyViewedProductsSection from './RecentlyViewedProductsSection';
import themeStyles from './SimilarRecentlyViewed.styles';
import { SimilarRecentlyViewedProvider } from './SimilarRecentlyViewedContext';
import getContentMargins from '../../utils/getContentMargins';

const Container = themeStyles.apply(Box, 'Container');

const SimilarRecentlyViewedUI = ({
  params = {},
  gtmTags,
  relatedModels,
  similarModels,
  models,
  getModels,
  fetchFinancial,
}) => {
  const flexDesktopAlignment = relatedModels.length ? 'flex-start' : 'flex-end';
  const margins = getContentMargins(params);

  return (
    <Wrapper>
      <Container margins={margins} flexDesktopAlignment={flexDesktopAlignment}>
        <SimilarRecentlyViewedProvider>
          <SimilarProductsSection models={similarModels} gtmTags={gtmTags} fetchFinancial={fetchFinancial} />
          <RecentlyViewedProductsSection
            models={models}
            getModels={getModels}
            gtmTags={gtmTags}
            fetchFinancial={fetchFinancial}
          />
        </SimilarRecentlyViewedProvider>
      </Container>
    </Wrapper>
  );
};

SimilarRecentlyViewedUI.propTypes = {
  params: PropTypes.shape({}),
  rendering: PropTypes.shape({
    componentName: PropTypes.string,
  }),
  relatedModels: PropTypes.arrayOf(PropTypes.shape({})),
  similarModels: PropTypes.arrayOf(PropTypes.shape({})),
  models: PropTypes.arrayOf(PropTypes.shape({})),
  fetchFinancial: PropTypes.bool,
};

export default SimilarRecentlyViewedUI;
