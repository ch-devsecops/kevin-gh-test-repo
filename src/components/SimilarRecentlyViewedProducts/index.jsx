import React from 'react';
import PropTypes from 'prop-types';
import { useSitecoreContext } from '@sitecore-jss/sitecore-jss-react';
import SimilarRecentlyViewedUI from './SimilarRecentlyViewedUI';
import NotImplemented from '../NotImplemented';
import { ENGINE_SITE_NAME, EngineSeriesCard, MARINE_SITE_NAME } from '../../utils/constants';
import { getEngineModels, getMarineModels } from './utils';
import getValueByKey from '../../utils/getValueByKey';

const SimilarRecentlyViewedMap = ({ fields, rendering, ...rest }) => {
  const {
    sitecoreContext: {
      site: { name: appName },
    },
  } = useSitecoreContext();

  if (!fields) return null;

  const value = fields?.data?.value;
  const catalog = fields?.data?.catalog;
  const relatedModels = getValueByKey(value, 'item');
  const relatedModelsDetIds = relatedModels?.map(model => model?.detIdentifier?.value);

  const componentName = rendering?.componentName;
  const gtmTags = {
    componentName,
    title: EngineSeriesCard?.Clicks?.DATA_GTM_TITLE,
    compareTitle: EngineSeriesCard?.Compare?.DATA_GTM_TITLE,
    interactionType: EngineSeriesCard?.Clicks?.DATA_GTM_INTERACTION_TYPE,
    compareInteractionType: EngineSeriesCard?.Compare?.DATA_GTM_INTERACTION_TYPE,
  };

  let models;
  let getModels;
  let similarModels;
  let fetchFinancial;

  switch (appName) {
    case ENGINE_SITE_NAME:
      models = getValueByKey(catalog, 'series');
      similarModels = getEngineModels(relatedModelsDetIds, models);
      getModels = getEngineModels;
      fetchFinancial = false;
      break;

    case MARINE_SITE_NAME:
      models = getValueByKey(catalog, 'outboards');
      similarModels = getMarineModels(relatedModelsDetIds, models);
      getModels = getMarineModels;
      fetchFinancial = true;
      break;

    default:
      return <NotImplemented name="SimilarRecentlyViewedProducts" />;
  }

  return (
    <SimilarRecentlyViewedUI
      models={models}
      similarModels={similarModels}
      relatedModels={relatedModels}
      getModels={getModels}
      fetchFinancial={fetchFinancial}
      gtmTags={gtmTags}
      {...rest}
    />
  );
};

SimilarRecentlyViewedMap.propTypes = {
  fields: PropTypes.shape({}),
};

export default SimilarRecentlyViewedMap;
