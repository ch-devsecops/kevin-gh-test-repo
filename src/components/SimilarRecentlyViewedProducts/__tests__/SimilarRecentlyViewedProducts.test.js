import React from 'react';
import { axe, cleanup, jssRender } from 'test-utils';
import { SitecoreContext } from '@sitecore-jss/sitecore-jss-react';
import SimilarRecentlyViewedProducts from '..';
import { ENGINE_SITE_NAME } from '../../../utils/constants';

afterEach(cleanup);

const mockData = require('../../../../sample-mock-data/routes/similar-recently-viewed-products/en.json');
const emptyMockData = require('../../../../sample-mock-data/routes/similar-recently-viewed-products/empty_en.json');

const contextMock = {
  site: {
    name: 'Engine',
  },
  pageState: 'normal',
  hondaRestApiHost: 'uat-api.honda.ca',
};

describe('<SimilarRecentlyViewedProducts>', () => {
  test('should render correctly', () => {
    const { container } = jssRender(
      <SitecoreContext
        context={{
          language: 'en',
          site: { name: ENGINE_SITE_NAME },
          ...contextMock,
        }}
        componentFactory={{}}
      >
        <SimilarRecentlyViewedProducts fields={mockData?.fields} />
      </SitecoreContext>,
    );

    expect(container).toMatchSnapshot();
    expect(container.children[0]).toBeTruthy();
  });

  test('should return null if there is no fields', () => {
    const { container } = jssRender(<SimilarRecentlyViewedProducts fields={null} />);

    expect(container).toMatchSnapshot();
  });

  test('should return null if the data is empty', () => {
    const { container } = jssRender(
      <SitecoreContext
        context={{
          language: 'en',
          site: { name: ENGINE_SITE_NAME },
          ...contextMock,
        }}
        componentFactory={{}}
      >
        <SimilarRecentlyViewedProducts fields={emptyMockData?.fields} />
      </SitecoreContext>,
    );
    expect(container).toMatchSnapshot();
  });

  test('should have no accessibility violations', async () => {
    const { container } = jssRender(
      <SitecoreContext
        context={{
          language: 'en',
          site: { name: ENGINE_SITE_NAME },
          ...contextMock,
        }}
        componentFactory={{}}
      >
        <SimilarRecentlyViewedProducts fields={mockData?.fields} />
      </SitecoreContext>,
    );

    expect(container).toMatchSnapshot();
    expect(await axe(container)).toHaveNoViolations();
  });
});
