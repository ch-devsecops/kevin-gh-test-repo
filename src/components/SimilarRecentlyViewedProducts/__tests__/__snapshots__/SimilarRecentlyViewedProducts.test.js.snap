// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SimilarRecentlyViewedProducts> should have no accessibility violations 1`] = `
<div>
  <div
    class="Box-sc-1bsf784-0 Wrapper-n67dtg-0 gXilRl"
  >
    <div
      class="Box-sc-1bsf784-0 sc-1k44dfv-0 dNDcUp"
    >
      <div
        class="Box-sc-1bsf784-0 sc-1k44dfv-0 hwpdqT"
        display="flex"
        width="1,1,0.5"
      >
        <h4
          aria-label="Pages.Models.Exploration.similarProductsLabel"
          class="sc-1k44dfv-0 chqxmS"
        >
          Pages.Models.Exploration.similarProductsLabel
        </h4>
        <div
          class="Box-sc-1bsf784-0 sc-1k44dfv-0 heIgvm"
        >
          <div
            class="Box-sc-1bsf784-0 sc-1k44dfv-0 fZLgxb"
          >
            <div
              class="Box-sc-1bsf784-0 sc-1k44dfv-0 ljGqou"
              data-testid="product-card"
            >
              <div
                style="overflow: hidden; transition-property: height; display: block; transition-duration: 0.25s;"
              />
              <div
                style="overflow: hidden; transition-property: height; display: block; transition-duration: 0.25s;"
              >
                <a
                  class="Link-sc-16wd4a0-0 bXBGRT"
                  data-gtm-body-style="Mini 4-Stroke"
                  data-gtm-interaction-type="explore"
                  data-gtm-model="HorizontalCrankshaft"
                  data-gtm-trim="GX35"
                  href="/en/mini_4_stroke/gx35"
                  title="view details"
                >
                  <div
                    class="sc-1k44dfv-0 cALsOM"
                    mb="xs,s"
                    mt="0"
                  >
                    

                    <h5
                      class="sc-1k44dfv-0 gXWcpP"
                    >
                      GX35
                    </h5>
                  </div>
                </a>
              </div>
              <a
                class="Link-sc-16wd4a0-0 bXBGRT"
                data-gtm-body-style="Mini 4-Stroke"
                data-gtm-interaction-type="explore"
                data-gtm-model="HorizontalCrankshaft"
                data-gtm-trim="GX35"
                href="/en/mini_4_stroke/gx35"
                title="view details"
              >
                <div
                  class="Box-sc-1bsf784-0 jgcAXQ"
                  height="139px,248px"
                  order="-1"
                >
                  <img
                    alt="Photo of Honda GX35 engine"
                    class="sc-1k44dfv-0 jLTFZk"
                    draggable="true"
                    loading="lazy"
                    src="sample-mock-data/media/img/product-card/engineImage.png"
                    style="object-fit: contain;"
                  />
                </div>
              </a>
              <div
                class="Box-sc-1bsf784-0 fznlGo"
              >
                <div
                  class="sc-1k44dfv-0 ktMwmo"
                >
                  

                  <p>
                    Make light work of hard tasks, while still getting cleaner, more efficient performance for a wide range of equipment.
                  </p>
                </div>
              </div>
              <a
                class="Button-pulw6t-0 sc-1k44dfv-0 eIRfAq"
                data-gtm-body-style="Mini 4-Stroke"
                data-gtm-interaction-type="explore"
                data-gtm-model="HorizontalCrankshaft"
                data-gtm-title="view details"
                data-gtm-trim="GX35"
                data-testid="dsr-button-component"
                display="none,block"
                href="/en/mini_4_stroke/gx35"
                tabindex="0"
              >
                Shared.Common.viewDetailsButton
              </a>
              <div
                class="Box-sc-1bsf784-0 sc-1k44dfv-0 jkGzMS"
              >
                <label
                  class="sc-bwzfXH cvvuMq"
                  data-gtm-body-style="Mini 4-Stroke"
                  data-gtm-interaction-type="compare item click"
                  data-gtm-model="HorizontalCrankshaft"
                  data-gtm-title="added comparison item"
                  data-gtm-trim="GX35"
                  data-testid="product-card-compare-btn"
                  data-testvalue="false"
                >
                  <input
                    aria-checked="false"
                    aria-label="Shared.Common.compareCheckboxAria"
                    autocomplete="off"
                    class="sc-bdVaJa gBQUrE"
                    data-gtm-body-style="Mini 4-Stroke"
                    data-gtm-interaction-type="compare item click"
                    data-gtm-model="HorizontalCrankshaft"
                    data-gtm-title="added comparison item"
                    data-gtm-trim="GX35"
                    id="7282790A8D9F433E8870E06231656A0C"
                    tabindex="-1"
                    type="checkbox"
                  />
                  <span
                    class="sc-htpNat hVaxDi"
                    data-testid="cy-compare-toggle"
                  />
                  <label
                    class="Label-sc-115wxjh-0 sc-1k44dfv-0 jLfsOv"
                    for="7282790A8D9F433E8870E06231656A0C"
                  >
                    Shared.Common.addToCompareButton
                  </label>
                </label>
              </div>
            </div>
          </div>
          <div
            class="Box-sc-1bsf784-0 sc-1k44dfv-0 fZLgxb"
          >
            <div
              class="Box-sc-1bsf784-0 sc-1k44dfv-0 ljGqou"
              data-testid="product-card"
            >
              <div
                style="overflow: hidden; transition-property: height; display: block; transition-duration: 0.25s;"
              />
              <div
                style="overflow: hidden; transition-property: height; display: block; transition-duration: 0.25s;"
              >
                <a
                  class="Link-sc-16wd4a0-0 bXBGRT"
                  data-gtm-body-style="Mini 4-Stroke"
                  data-gtm-interaction-type="explore"
                  data-gtm-model="HorizontalCrankshaft"
                  data-gtm-trim="GX50"
                  href="/en/mini_4_stroke/gx50"
                  title="view details"
                >
                  <div
                    class="sc-1k44dfv-0 cALsOM"
                    mb="xs,s"
                    mt="0"
                  >
                    

                    <h5
                      class="sc-1k44dfv-0 gXWcpP"
                    >
                      GX50
                    </h5>
                  </div>
                </a>
              </div>
              <a
                class="Link-sc-16wd4a0-0 bXBGRT"
                data-gtm-body-style="Mini 4-Stroke"
                data-gtm-interaction-type="explore"
                data-gtm-model="HorizontalCrankshaft"
                data-gtm-trim="GX50"
                href="/en/mini_4_stroke/gx50"
                title="view details"
              >
                <div
                  class="Box-sc-1bsf784-0 jgcAXQ"
                  height="139px,248px"
                  order="-1"
                >
                  <img
                    alt="Photo of Honda GX50 engine"
                    class="sc-1k44dfv-0 jLTFZk"
                    draggable="true"
                    loading="lazy"
                    src="sample-mock-data/media/img/product-card/engineImage.png"
                    style="object-fit: contain;"
                  />
                </div>
              </a>
              <div
                class="Box-sc-1bsf784-0 fznlGo"
              >
                <div
                  class="sc-1k44dfv-0 ktMwmo"
                >
                  

                  <p>
                    Professionals trust the GX50 to deliver a solid performance across an impressive range of applications from commercial gardening to construction and agriculture.
                  </p>
                </div>
              </div>
              <a
                class="Button-pulw6t-0 sc-1k44dfv-0 eIRfAq"
                data-gtm-body-style="Mini 4-Stroke"
                data-gtm-interaction-type="explore"
                data-gtm-model="HorizontalCrankshaft"
                data-gtm-title="view details"
                data-gtm-trim="GX50"
                data-testid="dsr-button-component"
                display="none,block"
                href="/en/mini_4_stroke/gx50"
                tabindex="0"
              >
                Shared.Common.viewDetailsButton
              </a>
              <div
                class="Box-sc-1bsf784-0 sc-1k44dfv-0 jkGzMS"
              >
                <label
                  class="sc-bwzfXH cvvuMq"
                  data-gtm-body-style="Mini 4-Stroke"
                  data-gtm-interaction-type="compare item click"
                  data-gtm-model="HorizontalCrankshaft"
                  data-gtm-title="added comparison item"
                  data-gtm-trim="GX50"
                  data-testid="product-card-compare-btn"
                  data-testvalue="false"
                >
                  <input
                    aria-checked="false"
                    aria-label="Shared.Common.compareCheckboxAria"
                    autocomplete="off"
                    class="sc-bdVaJa gBQUrE"
                    data-gtm-body-style="Mini 4-Stroke"
                    data-gtm-interaction-type="compare item click"
                    data-gtm-model="HorizontalCrankshaft"
                    data-gtm-title="added comparison item"
                    data-gtm-trim="GX50"
                    id="8E16B45522284B50AC1868819AA989BC"
                    tabindex="-1"
                    type="checkbox"
                  />
                  <span
                    class="sc-htpNat hVaxDi"
                    data-testid="cy-compare-toggle"
                  />
                  <label
                    class="Label-sc-115wxjh-0 sc-1k44dfv-0 jLfsOv"
                    for="8E16B45522284B50AC1868819AA989BC"
                  >
                    Shared.Common.addToCompareButton
                  </label>
                </label>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`<SimilarRecentlyViewedProducts> should render correctly 1`] = `
<div>
  <div
    class="Box-sc-1bsf784-0 Wrapper-n67dtg-0 gXilRl"
  >
    <div
      class="Box-sc-1bsf784-0 sc-1k44dfv-0 dNDcUp"
    >
      <div
        class="Box-sc-1bsf784-0 sc-1k44dfv-0 hwpdqT"
        display="flex"
        width="1,1,0.5"
      >
        <h4
          aria-label="Pages.Models.Exploration.similarProductsLabel"
          class="sc-1k44dfv-0 chqxmS"
        >
          Pages.Models.Exploration.similarProductsLabel
        </h4>
        <div
          class="Box-sc-1bsf784-0 sc-1k44dfv-0 heIgvm"
        >
          <div
            class="Box-sc-1bsf784-0 sc-1k44dfv-0 fZLgxb"
          >
            <div
              class="Box-sc-1bsf784-0 sc-1k44dfv-0 ljGqou"
              data-testid="product-card"
            >
              <div
                style="overflow: hidden; transition-property: height; display: block; transition-duration: 0.25s;"
              />
              <div
                style="overflow: hidden; transition-property: height; display: block; transition-duration: 0.25s;"
              >
                <a
                  class="Link-sc-16wd4a0-0 bXBGRT"
                  data-gtm-body-style="Mini 4-Stroke"
                  data-gtm-interaction-type="explore"
                  data-gtm-model="HorizontalCrankshaft"
                  data-gtm-trim="GX35"
                  href="/en/mini_4_stroke/gx35"
                  title="view details"
                >
                  <div
                    class="sc-1k44dfv-0 cALsOM"
                    mb="xs,s"
                    mt="0"
                  >
                    

                    <h5
                      class="sc-1k44dfv-0 gXWcpP"
                    >
                      GX35
                    </h5>
                  </div>
                </a>
              </div>
              <a
                class="Link-sc-16wd4a0-0 bXBGRT"
                data-gtm-body-style="Mini 4-Stroke"
                data-gtm-interaction-type="explore"
                data-gtm-model="HorizontalCrankshaft"
                data-gtm-trim="GX35"
                href="/en/mini_4_stroke/gx35"
                title="view details"
              >
                <div
                  class="Box-sc-1bsf784-0 jgcAXQ"
                  height="139px,248px"
                  order="-1"
                >
                  <img
                    alt="Photo of Honda GX35 engine"
                    class="sc-1k44dfv-0 jLTFZk"
                    draggable="true"
                    loading="lazy"
                    src="sample-mock-data/media/img/product-card/engineImage.png"
                    style="object-fit: contain;"
                  />
                </div>
              </a>
              <div
                class="Box-sc-1bsf784-0 fznlGo"
              >
                <div
                  class="sc-1k44dfv-0 ktMwmo"
                >
                  

                  <p>
                    Make light work of hard tasks, while still getting cleaner, more efficient performance for a wide range of equipment.
                  </p>
                </div>
              </div>
              <a
                class="Button-pulw6t-0 sc-1k44dfv-0 eIRfAq"
                data-gtm-body-style="Mini 4-Stroke"
                data-gtm-interaction-type="explore"
                data-gtm-model="HorizontalCrankshaft"
                data-gtm-title="view details"
                data-gtm-trim="GX35"
                data-testid="dsr-button-component"
                display="none,block"
                href="/en/mini_4_stroke/gx35"
                tabindex="0"
              >
                Shared.Common.viewDetailsButton
              </a>
              <div
                class="Box-sc-1bsf784-0 sc-1k44dfv-0 jkGzMS"
              >
                <label
                  class="sc-bwzfXH cvvuMq"
                  data-gtm-body-style="Mini 4-Stroke"
                  data-gtm-interaction-type="compare item click"
                  data-gtm-model="HorizontalCrankshaft"
                  data-gtm-title="added comparison item"
                  data-gtm-trim="GX35"
                  data-testid="product-card-compare-btn"
                  data-testvalue="false"
                >
                  <input
                    aria-checked="false"
                    aria-label="Shared.Common.compareCheckboxAria"
                    autocomplete="off"
                    class="sc-bdVaJa gBQUrE"
                    data-gtm-body-style="Mini 4-Stroke"
                    data-gtm-interaction-type="compare item click"
                    data-gtm-model="HorizontalCrankshaft"
                    data-gtm-title="added comparison item"
                    data-gtm-trim="GX35"
                    id="7282790A8D9F433E8870E06231656A0C"
                    tabindex="-1"
                    type="checkbox"
                  />
                  <span
                    class="sc-htpNat hVaxDi"
                    data-testid="cy-compare-toggle"
                  />
                  <label
                    class="Label-sc-115wxjh-0 sc-1k44dfv-0 jLfsOv"
                    for="7282790A8D9F433E8870E06231656A0C"
                  >
                    Shared.Common.addToCompareButton
                  </label>
                </label>
              </div>
            </div>
          </div>
          <div
            class="Box-sc-1bsf784-0 sc-1k44dfv-0 fZLgxb"
          >
            <div
              class="Box-sc-1bsf784-0 sc-1k44dfv-0 ljGqou"
              data-testid="product-card"
            >
              <div
                style="overflow: hidden; transition-property: height; display: block; transition-duration: 0.25s;"
              />
              <div
                style="overflow: hidden; transition-property: height; display: block; transition-duration: 0.25s;"
              >
                <a
                  class="Link-sc-16wd4a0-0 bXBGRT"
                  data-gtm-body-style="Mini 4-Stroke"
                  data-gtm-interaction-type="explore"
                  data-gtm-model="HorizontalCrankshaft"
                  data-gtm-trim="GX50"
                  href="/en/mini_4_stroke/gx50"
                  title="view details"
                >
                  <div
                    class="sc-1k44dfv-0 cALsOM"
                    mb="xs,s"
                    mt="0"
                  >
                    

                    <h5
                      class="sc-1k44dfv-0 gXWcpP"
                    >
                      GX50
                    </h5>
                  </div>
                </a>
              </div>
              <a
                class="Link-sc-16wd4a0-0 bXBGRT"
                data-gtm-body-style="Mini 4-Stroke"
                data-gtm-interaction-type="explore"
                data-gtm-model="HorizontalCrankshaft"
                data-gtm-trim="GX50"
                href="/en/mini_4_stroke/gx50"
                title="view details"
              >
                <div
                  class="Box-sc-1bsf784-0 jgcAXQ"
                  height="139px,248px"
                  order="-1"
                >
                  <img
                    alt="Photo of Honda GX50 engine"
                    class="sc-1k44dfv-0 jLTFZk"
                    draggable="true"
                    loading="lazy"
                    src="sample-mock-data/media/img/product-card/engineImage.png"
                    style="object-fit: contain;"
                  />
                </div>
              </a>
              <div
                class="Box-sc-1bsf784-0 fznlGo"
              >
                <div
                  class="sc-1k44dfv-0 ktMwmo"
                >
                  

                  <p>
                    Professionals trust the GX50 to deliver a solid performance across an impressive range of applications from commercial gardening to construction and agriculture.
                  </p>
                </div>
              </div>
              <a
                class="Button-pulw6t-0 sc-1k44dfv-0 eIRfAq"
                data-gtm-body-style="Mini 4-Stroke"
                data-gtm-interaction-type="explore"
                data-gtm-model="HorizontalCrankshaft"
                data-gtm-title="view details"
                data-gtm-trim="GX50"
                data-testid="dsr-button-component"
                display="none,block"
                href="/en/mini_4_stroke/gx50"
                tabindex="0"
              >
                Shared.Common.viewDetailsButton
              </a>
              <div
                class="Box-sc-1bsf784-0 sc-1k44dfv-0 jkGzMS"
              >
                <label
                  class="sc-bwzfXH cvvuMq"
                  data-gtm-body-style="Mini 4-Stroke"
                  data-gtm-interaction-type="compare item click"
                  data-gtm-model="HorizontalCrankshaft"
                  data-gtm-title="added comparison item"
                  data-gtm-trim="GX50"
                  data-testid="product-card-compare-btn"
                  data-testvalue="false"
                >
                  <input
                    aria-checked="false"
                    aria-label="Shared.Common.compareCheckboxAria"
                    autocomplete="off"
                    class="sc-bdVaJa gBQUrE"
                    data-gtm-body-style="Mini 4-Stroke"
                    data-gtm-interaction-type="compare item click"
                    data-gtm-model="HorizontalCrankshaft"
                    data-gtm-title="added comparison item"
                    data-gtm-trim="GX50"
                    id="8E16B45522284B50AC1868819AA989BC"
                    tabindex="-1"
                    type="checkbox"
                  />
                  <span
                    class="sc-htpNat hVaxDi"
                    data-testid="cy-compare-toggle"
                  />
                  <label
                    class="Label-sc-115wxjh-0 sc-1k44dfv-0 jLfsOv"
                    for="8E16B45522284B50AC1868819AA989BC"
                  >
                    Shared.Common.addToCompareButton
                  </label>
                </label>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`<SimilarRecentlyViewedProducts> should return null if the data is empty 1`] = `
<div>
  <div
    class="Box-sc-1bsf784-0 Wrapper-n67dtg-0 gXilRl"
  >
    <div
      class="Box-sc-1bsf784-0 sc-1k44dfv-0 gqakPH"
    />
  </div>
</div>
`;

exports[`<SimilarRecentlyViewedProducts> should return null if there is no fields 1`] = `<div />`;
